database/migrations/2024_06_11_231143_add_rotessa_gateway.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Support\Str;
+use App\Models\Gateway;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Model::unguard();
+        \DB::statement('SET FOREIGN_KEY_CHECKS=0;');
```

Comment:
no need for the foreign key checks here.

also, you may want to use 

database/migrations/2023_06_20_123355_add_paypal_rest_payment_driver.php

as a template, the gateway->key field needs to be a static string, the one in this migration will create a new static string per installation. we do use the ->key elsewhere in the application, so it needs to be uniform.

you will also need to add the gateway array to the PaymentLibrarySeeders.php

Reply:
Refactored as per request and added to payment library seeder